lib/ansible/modules/cloud/amazon/cloudformation_stack_set.py

```diff
@@ -0,0 +1,660 @@
+    # automatic backoff / retry for throttling error codes
+    cfn = module.client('cloudformation', retry_decorator=AWSRetry.jittered_backoff(retries=10, delay=3, max_delay=30))
+    existing_stack_set = stack_set_facts(cfn, module.params['name'])
+
+    operation_uuid = to_native(uuid.uuid4())
+    operation_ids = []
+    # collect the parameters that are passed to boto3. Keeps us from having so many scalars floating around.
+    stack_params = {}
+    state = module.params['state']
+    if state == 'present' and not module.params['accounts']:
+        module.fail_json(
+            msg="Can't create a stack set without choosing at least one account. "
+                "To get the ID of the current account, use the aws_caller_facts module."
+        )
+
+    module.params['accounts'] = [to_native(a) for a in module.params['accounts']]
+
+    stack_params['StackSetName'] = module.params['name']
+    if module.params.get('description'):
+        stack_params['Description'] = module.params['description']
+
+    if module.params.get('capabilities'):
+        stack_params['Capabilities'] = module.params['capabilities']
+
+    if module.params['template'] is not None:
+        with open(module.params['template'], 'r') as tpl:
+            stack_params['TemplateBody'] = tpl.read()
+    elif module.params['template_body'] is not None:
+        stack_params['TemplateBody'] = module.params['template_body']
+    elif module.params['template_url'] is not None:
+        stack_params['TemplateURL'] = module.params['template_url']
+    else:
+        # no template is provided, but if the stack set exists already, we can use the existing one.
+        if existing_stack_set:
+            stack_params['UsePreviousTemplate'] = True
+        else:
+            module.fail_json(
+                msg="The Stack Set {0} does not exist, and no template was provided. Provide one of `template`, "
+                    "`template_body`, or `template_url`".format(module.params['name'])
+            )
+
+    stack_params['Parameters'] = []
+    for k, v in module.params['parameters'].items():
+        if isinstance(v, dict):
+            # set parameter based on a dict to allow additional CFN Parameter Attributes
+            param = dict(ParameterKey=k)
+
+            if 'value' in v:
+                param['ParameterValue'] = to_native(v['value'])
+
+            if 'use_previous_value' in v and bool(v['use_previous_value']):
+                param['UsePreviousValue'] = True
+                param.pop('ParameterValue', None)
+
+            stack_params['Parameters'].append(param)
+        else:
+            # allow default k/v configuration to set a template parameter
+            stack_params['Parameters'].append({'ParameterKey': k, 'ParameterValue': str(v)})
+
+    if module.params.get('tags') and isinstance(module.params.get('tags'), dict):
+        stack_params['Tags'] = ansible.module_utils.ec2.ansible_dict_to_boto3_tag_list(module.params['tags'])
```

Comment:
You also imported ansible_dict_to_boto3_tag_list. There's a note by the ec2 import:
```
import ansible.module_utils.ec2
# import a class, otherwise we'll use a fully qualified path
```
Is that old?

Reply:
Yeah, whoops that's not needed.